system/jlib/jlog.cpp

```diff
@@ -3488,6 +3488,95 @@ IRemoteLogAccess *queryRemoteLogAccessor()
     );
 }
 
+void diagnoseLogAccessPluginLoad(LogAccessPluginDiagnostics & diagnostics)
+{
+    constexpr const char * methodName = "diagnoseLogAccessPluginLoad";
+    constexpr const char * instFactoryName = "createInstance";
+
+    try
+    {
+        Owned<IPropertyTree> logAccessPluginConfig = getGlobalConfigSP()->getPropTree("logAccess");
+
+        if (!logAccessPluginConfig)
+        {
+            diagnostics.configFound = false;
+            diagnostics.statusMessage.set("No logaccess configuration found in global config");
+            return;
+        }
+
+        diagnostics.configFound = true;
+
+        StringBuffer type;
+        logAccessPluginConfig->getProp("@type", type);
+        if (type.isEmpty())
+        {
+            diagnostics.statusMessage.set("RemoteLogAccess plugin type not specified in configuration");
+            return;
+        }
+
+        diagnostics.pluginType.set(type.str());
```

Comment:
Why not just set diagnostics.pluginType then check for empty? More importantly, is the local type copy necessary?